Libraries/Core/ReactNativeVersion.js

```diff
@@ -1,16 +1,16 @@
 /**
+ * @generated by scripts/set-rn-version.js
+ *
  * Copyright (c) Meta Platforms, Inc. and affiliates.
  *
  * This source code is licensed under the MIT license found in the
  * LICENSE file in the root directory of this source tree.
  *
- * @format
- * @generated by scripts/set-rn-version.js
  * @flow strict
  */
 
 exports.version = {
-  major: 0,
+  major: 1000,
```

Comment:
Change looks fine to me. Let's just make sure that the nightly and the commitly jobs are effectively changing this to `0`. I'd like to avoid to publish `1000.0.0-20220928-2025-527cc3426` on NPM just because the script it's appending the date/sha 👍

Reply:
we should be fine, the flag sets the version to 0.0.0 here: https://github.com/facebook/react-native/blob/49b14cc6033db2d5561e2cd1094e220bf256bf0a/scripts/publish-npm.js#L152

and then releaseversion gets modified here: https://github.com/facebook/react-native/blob/49b14cc6033db2d5561e2cd1094e220bf256bf0a/scripts/publish-npm.js#L183

Reply:
but that said, for commitlies it's not setting it to 0.0.0 - uh.

We might need to change this https://github.com/facebook/react-native/blob/49b14cc6033db2d5561e2cd1094e220bf256bf0a/scripts/publish-npm.js#L157

to use `isCommitly`? 🤔
Why are commitlies not set to 0.0.0? can they be happening on stable branches?
cc @lunaleaps ?

Reply:
Hmm trying to page this all back in. Yes I believe commitlies can happen on stable branches -- I think the idea was that we could have PRs against the stable branch if people wanted to release a new patch -- and could use the commitly to verify. 

I think the 0.0.0 versioning is just that we could publish nightlies and it wouldn't affect npm version ordering but I think commitlies probably were grandfathered in to keep the `1000` version -- I don't think it was an intentional thing. I think we could probably update everything to `0.0.0`

Reply:
actually, I'm stupid - I should have read the comment at the start of the file: https://github.com/facebook/react-native/blob/49b14cc6033db2d5561e2cd1094e220bf256bf0a/scripts/publish-npm.js#L17

looks like currently we do have commitlies in the 1000.0.0 format - @cortinico I don't know if we want to move away from the pre-established approach.

Reply:
> looks like currently we do have commitlies in the 1000.0.0 format - @cortinico I don't know if we want to move away from the pre-established approach.

Ideally yes. I would refrain from packaging anythign on `1000.x` versioning scheme.
Commitlies are just nightlies from my point of view (with a different date/time and sha) + we don't `npm publish` them. Aside from this, versioning schema should be the same.

As @lunaleaps, commitlies are avaialble on every commit, also release branches. In an ideal world, release managers should test the commitly before triggering the publish job.

Reply:
ok so just to make sure I'm understanding  - I should change the logic here to tag/version commitlies to use 0.0.0, correct?

Reply:
It doesn't necessarily need to happen here. Can be a separate PR like "re-align commitlies versioning schema to follow the same as nightlies" or so 👍